### DISCUSSION TITLE
Getting the ClientContext from PnPContext

### DISCUSSION THREAD

**Discussion by eduardpaul** [pnp/pnpcore#476] [Q&A] [answered]
Hi all, I am trying to get the ClientContext from PnPContext as indicated in the documentation [here](https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html#using-pnp-framework-when-the-pnp-core-sdk-was-already-configured), but PnPCoreSdk singleton does not have any GetContext or GetClientContext. 

![image](https://user-images.githubusercontent.com/1370426/121008882-c5590e80-c793-11eb-85aa-a3cd7ae4507a.png)

I'm not sure if it's a documentation issue or a bug but i couldn't find any reference of this in pnpcore source code; only an old [test case](https://github.com/pnp/pnpframework/blob/2b7999d2605aa312fa6846a34e91515f859d456a/src/lib/PnP.Framework.Test/PnPCoreSdk/PnPCoreSdkTests.cs).

It would be nice to hace this method back or an alternative. 
Thanks. 

Update: after some searching, my current solution is to use provider's access token as follow: 

`

        public ClientContext GetClientContext(string siteUrl)
        {
            var provider = new X509CertificateAuthenticationProvider(azureFunctionSettings.ClientId,
                                azureFunctionSettings.TenantId,
                                StoreName.My,
                                StoreLocation.CurrentUser,
                                azureFunctionSettings.CertificateThumbprint);

            var clientContext = new ClientContext(siteUrl)
            {
                DisableReturnValueCache = true
            };
            
            clientContext.ExecutingWebRequest += (sender, args) =>
            {
                Uri resourceUri = new Uri(siteUrl);
                resourceUri = new Uri(resourceUri.Scheme + "://" + resourceUri.Host + "/");

                string accessToken = provider.GetAccessTokenAsync(resourceUri).Result;
                args.WebRequestExecutor.RequestHeaders["Authorization"] = "Bearer " + accessToken;
            };

            return clientContext;
        }

`

**Comment by DaleyKD** [comment, marked as answer]
@eduardpaul : Are you using version 1.1.0 or a nightly?  (IOW, what version of the NuGet did you get?)

**Reply by DaleyKD** [reply]
UPDATE:  As of like 30 minutes ago, it's now version 1.2.0!  Whoop, whoop!

**Reply by eduardpaul** [reply]
I'm using the following: 
 `
    <PackageReference Include="PnP.Core" Version="1.1.0" />
    <PackageReference Include="PnP.Core.Auth" Version="1.1.0" />
    <PackageReference Include="PnP.Framework" Version="1.4.0" />
`

**Reply by DaleyKD** [reply]
@eduardpaul : Update those to the latest.  That should get you going.  A LOT has changed since 1.1.0. 😄

**Reply by eduardpaul** [reply]
@DaleyKD 🥇  updated PnP.Framework to 1.5 (published **2hours ago**) and  GetClientContext showed up .

**Reply by DaleyKD** [reply]
Sweet.  I don't know how, but mark it answered and have an awesome week. 👍

**Reply by eduardpaul** [reply]
Confirmed that it's not related to pnp core. It was a missing feature from pnpframework that was added this 1.5 release. It's on their changelog: 
![image](https://user-images.githubusercontent.com/1370426/121078194-eb54d200-c7d8-11eb-9b5c-26d83c2ef0a7.png)

it's funny though that it has been all this time in the pnpcore documentation... 

thank you very much for your time @DaleyKD !
